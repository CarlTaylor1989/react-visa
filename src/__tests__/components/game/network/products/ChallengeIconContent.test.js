/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import {
  ChallengeIconContent,
  mapStateToProps
} from '../../../../../components/game/network/products/ChallengeIconContent';
import {
  CHALLENGE_PERFECT_STATUS,
  CHALLENGE_COMPLETION_STATUS,
  CHALLENGE_AVAILABLE_STATUS,
  CHALLENGE_UNAVAILABLE_STATUS
} from '../../../../../config/constants';


let mockProps;
beforeEach(() => {
  mockProps = {
    challenges: [],
    challengeId: 'c1',
    coordinates: [100, 100],
    familyId: 'm1f1',
    lastChallengesModified: [],
    currentProduct: '',
    mapInitialised: false,
    type: 'type1'
  };
});

test('renders the ChallengeIconContent component with the available state', () => {
  const wrapper = shallow(<ChallengeIconContent {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('method componentDidMount()', () => {
  const wrapper = shallow(<ChallengeIconContent {...mockProps} />, {
    disableLifecycleMethods: true
  });
  wrapper.instance().updateChallengeState = jest.fn();
  wrapper.instance().componentDidMount();
  expect(wrapper.instance().updateChallengeState).toHaveBeenCalled();
});

test('method componentDidUpdate()', () => {
  const wrapper = shallow(<ChallengeIconContent {...mockProps} />);
  wrapper.instance().updateChallengeState = jest.fn();
  wrapper.setProps({
    lastChallengesModified: ['c1', 'c2']
  });
  wrapper.update();
  expect(wrapper.instance().updateChallengeState).toHaveBeenCalled();
});

describe('method updateChallengeState()', () => {
  test('sets the status to completed', () => {
    const challenges = [{
      id: 'c1',
      familyId: 'm1f1',
      status: CHALLENGE_COMPLETION_STATUS
    }];
    const wrapper = shallow(<ChallengeIconContent {...mockProps} challenges={challenges} />);
    wrapper.instance().updateChallengeState();
    expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
    expect(wrapper.instance().state.status).toEqual('completed');
    expect(wrapper.instance().state.challengePoints).toEqual(expect.any(Number));
  });

  test('sets the status to perfect', () => {
    const challenges = [{
      id: 'c1',
      familyId: 'm1f1',
      status: CHALLENGE_PERFECT_STATUS
    }];
    const wrapper = shallow(<ChallengeIconContent {...mockProps} challenges={challenges} />);
    wrapper.instance().updateChallengeState();
    expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
    expect(wrapper.instance().state.status).toEqual('perfect');
    expect(wrapper.instance().state.challengePoints).toEqual(expect.any(Number));
  });

  test('sets the status to available', () => {
    const challenges = [{
      id: 'c1',
      familyId: 'm1f1',
      status: CHALLENGE_AVAILABLE_STATUS
    }];
    const wrapper = shallow(<ChallengeIconContent {...mockProps} challenges={challenges} />);
    wrapper.instance().updateChallengeState();
    expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
    expect(wrapper.instance().state.status).toBe('available');
    expect(wrapper.instance().state.challengePoints).toEqual(expect.any(Number));
  });

  test('sets the status to available for the first tree node', () => {
    const challenges = [{
      id: 'c1',
      familyId: 'm1f1',
      status: CHALLENGE_UNAVAILABLE_STATUS,
      isFirst: true
    }];
    const wrapper = shallow(<ChallengeIconContent {...mockProps} challenges={challenges} />);
    wrapper.instance().updateChallengeState();
    expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
    expect(wrapper.instance().state.status).toBe('available');
    expect(wrapper.instance().state.challengePoints).toEqual(expect.any(Number));
  });
});

test('should mapStateToProps', () => {
  const state = {
    mapData: {
      challenges: [],
      initialised: false,
      lastChallengesModified: []
    }
  };
  const props = mapStateToProps(state);
  expect(props).toEqual({
    challenges: [],
    mapInitialised: false,
    lastChallengesModified: []
  });
});
