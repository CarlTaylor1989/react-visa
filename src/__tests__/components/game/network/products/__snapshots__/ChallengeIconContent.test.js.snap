// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`method updateChallengeState() sets the status to available 1`] = `
<div
  className="challengeIconContent type1"
  style={
    Object {
      "left": 100,
      "top": 100,
    }
  }
>
  <Translate
    className="statusText"
    tag="span"
    value="network.productpages.generic.available"
  />
  <span
    className="challengeIcon m1f1 available"
  />
  <div
    className="pointsContainer"
  >
    <div
      className="pointsText"
    >
      <Translate
        tag="span"
        value="network.productpages.generic.points"
      />
    </div>
    <NumberFormat
      allowEmptyFormatting={false}
      allowNegative={true}
      className="challengePoints"
      decimalSeparator="."
      displayType="text"
      fixedDecimalScale={false}
      getInputRef={[Function]}
      isAllowed={[Function]}
      isNumericString={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      onValueChange={[Function]}
      prefix=""
      suffix=""
      thousandSeparator=","
      thousandsGroupStyle="thousand"
      type="text"
      value={0}
    />
  </div>
</div>
`;

exports[`method updateChallengeState() sets the status to available for the first tree node 1`] = `
<div
  className="challengeIconContent type1"
  style={
    Object {
      "left": 100,
      "top": 100,
    }
  }
>
  <Translate
    className="statusText"
    tag="span"
    value="network.productpages.generic.available"
  />
  <span
    className="challengeIcon m1f1 available"
  />
  <div
    className="pointsContainer"
  >
    <div
      className="pointsText"
    >
      <Translate
        tag="span"
        value="network.productpages.generic.points"
      />
    </div>
    <NumberFormat
      allowEmptyFormatting={false}
      allowNegative={true}
      className="challengePoints"
      decimalSeparator="."
      displayType="text"
      fixedDecimalScale={false}
      getInputRef={[Function]}
      isAllowed={[Function]}
      isNumericString={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      onValueChange={[Function]}
      prefix=""
      suffix=""
      thousandSeparator=","
      thousandsGroupStyle="thousand"
      type="text"
      value={0}
    />
  </div>
</div>
`;

exports[`method updateChallengeState() sets the status to completed 1`] = `
<div
  className="challengeIconContent type1"
  style={
    Object {
      "left": 100,
      "top": 100,
    }
  }
>
  <Translate
    className="statusText"
    tag="span"
    value="network.productpages.generic.completed"
  />
  <span
    className="challengeIcon m1f1 completed"
  />
  <div
    className="pointsContainer"
  >
    <div
      className="pointsText"
    >
      <Translate
        tag="span"
        value="network.productpages.generic.points"
      />
    </div>
    <NumberFormat
      allowEmptyFormatting={false}
      allowNegative={true}
      className="challengePoints"
      decimalSeparator="."
      displayType="text"
      fixedDecimalScale={false}
      getInputRef={[Function]}
      isAllowed={[Function]}
      isNumericString={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      onValueChange={[Function]}
      prefix=""
      suffix=""
      thousandSeparator=","
      thousandsGroupStyle="thousand"
      type="text"
      value={10000}
    />
  </div>
</div>
`;

exports[`method updateChallengeState() sets the status to perfect 1`] = `
<div
  className="challengeIconContent type1"
  style={
    Object {
      "left": 100,
      "top": 100,
    }
  }
>
  <Translate
    className="statusText"
    tag="span"
    value="network.productpages.generic.perfect"
  />
  <span
    className="challengeIcon m1f1 perfect"
  />
  <div
    className="pointsContainer"
  >
    <div
      className="pointsText"
    >
      <Translate
        tag="span"
        value="network.productpages.generic.points"
      />
    </div>
    <NumberFormat
      allowEmptyFormatting={false}
      allowNegative={true}
      className="challengePoints"
      decimalSeparator="."
      displayType="text"
      fixedDecimalScale={false}
      getInputRef={[Function]}
      isAllowed={[Function]}
      isNumericString={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      onValueChange={[Function]}
      prefix=""
      suffix=""
      thousandSeparator=","
      thousandsGroupStyle="thousand"
      type="text"
      value={15000}
    />
  </div>
</div>
`;

exports[`renders the ChallengeIconContent component with the available state 1`] = `
<div
  className="challengeIconContent type1"
  style={
    Object {
      "left": 100,
      "top": 100,
    }
  }
>
  <Translate
    className="statusText"
    tag="span"
    value="network.productpages.generic.notStarted"
  />
  <span
    className="challengeIcon m1f1 notStarted"
  />
  <div
    className="pointsContainer"
  >
    <div
      className="pointsText"
    >
      <Translate
        tag="span"
        value="network.productpages.generic.points"
      />
    </div>
    <NumberFormat
      allowEmptyFormatting={false}
      allowNegative={true}
      className="challengePoints"
      decimalSeparator="."
      displayType="text"
      fixedDecimalScale={false}
      getInputRef={[Function]}
      isAllowed={[Function]}
      isNumericString={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      onValueChange={[Function]}
      prefix=""
      suffix=""
      thousandSeparator=","
      thousandsGroupStyle="thousand"
      type="text"
      value={0}
    />
  </div>
</div>
`;
